### PR TITLE
Enforce and document the fact that StorageDevice.format is always a DeviceFormat.

### PR DESCRIPTION
**Pull request by dwlehman**
StorageDevice.format should never be anything other than an instance
of DeviceFormat or one of its subclasses. All block devices contain data
that is arranged according to some format. The case of empty or
uninitialized or unrecognized formatting is represented by the base class
DeviceFormat.

### CODE REVIEW

File: blivet/devices/storage.py [hunk 1]
```diff
@@ -623,6 +643,14 @@ def _updateNetDevMountOption(self):
             self._format.options = ",".join(option_list)
 
     def _getFormat(self):
+        """ Get the device's format instance.
+
```
Comment (mulkieran) [inline blivet/devices/storage.py]: Might as well docstring it all up with ```:rtype:``` and ```:return:``` while you are here.

**Comment by mulkieran** [comment]
Other than neatpick is fine with me.

File: blivet/devices/storage.py [hunk 2]
```diff
@@ -592,9 +592,29 @@ def status(self):
...
+            A value of None will effectively mark the device as unformatted,
+            but this is accomplished by setting it to an instance of the base
+            :class:`~.formats.DeviceFormat` class.
+
+            .. note::
+                :attr:`format` should always be an instance of
+                :class:`~.formats.DeviceFormat`. To ensure this continues to be
+                the case, all subclasses that define their own :attr:`format`
+                getter should call :meth:`StorageDevice._setFormat` from their
+                getter.
```
Comment (vpodzime) [inline blivet/devices/storage.py]: Should this talk about **setter** rather than **getter**? If not, I'm a bit confused here.

Comment (dwlehman) [inline blivet/devices/storage.py]: Indeed it should. Thanks.

**Pull request merged**